Sources/News.php

```diff
@@ -1849,4 +2129,758 @@ function getXmlProfile($xml_format)
 	return $data;
 }
 
+/**
+ * Get a user's posts.
+ * The returned array will be generated to match the xml_format.
+ *
+ * @param string $xml_format The XML format. Can be 'atom', 'rdf', 'rss', 'rss2' or 'smf'
+ * @return array An array of arrays containing data for the feed. Each array has keys corresponding to the appropriate tags for the specified format.
+ */
+function getXmlPosts($xml_format)
+{
+	global $scripturl, $modSettings, $board, $txt, $context, $user_info;
+	global $query_this_board, $smcFunc, $sourcedir, $cachedir;
+
+	$uid = isset($_GET['u']) ? (int) $_GET['u'] : $user_info['id'];
+
+	if (empty($uid) || (!allowedTo('profile_view') && $uid != $user_info['id']))
+		return array();
+
+	$show_all = $user_info['is_admin'] || $uid == $user_info['id'];
+
+	// You are allowed in this special case to see your own posts from anywhere
+	if ($show_all)
+		$query_this_board = preg_replace('/\{query_see_board\}\s*(AND )?/', '', $query_this_board);
+
+	require_once($sourcedir . '/Subs-Attachments.php');
+
+	// Need to know the total so we can track our progress
+	if (!empty($context['batch_mode']) && empty($context['batch_total']))
+	{
+		$request = $smcFunc['db_query']('', '
+			SELECT COUNT(m.id_msg)
```

Comment:
you should safe this in a session var

Reply:
Perhaps a cache entry would be better? this way busy forums with custom cache settings will get uniform results for all their queries.

Reply:
Since session is safed in the database or in php fs,
i see no different beside that cache use some people and session use everyone.

Reply:
Yep, you're right, I forgot this is on an user level, thus a session would be better.